modules/local/cram_filter_minimap2_filter5end_fixmate_sort.nf

```diff
@@ -0,0 +1,59 @@
+process CRAM_FILTER_MINIMAP2_FILTER5END_FIXMATE_SORT {
+    tag "$meta.id"
+    label "process_high"
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-1a6fe65bd6674daba65066aa796ed8f5e8b4687b:688e175eb0db54de17822ba7810cc9e20fa06dd5-0' :
+        'biocontainers/mulled-v2-1a6fe65bd6674daba65066aa796ed8f5e8b4687b:688e175eb0db54de17822ba7810cc9e20fa06dd5-0' }"
+
+    input:
+    tuple val(meta), path(cramfile), path(cramindex), val(from), val(to), val(base), val(chunkid), val(rglines), val(ref), path(reference)
+
+    output:
+    tuple val(meta), path("*.bam"), emit: mappedbam
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args1 = task.ext.args1 ?: ''
+    def args2 = task.ext.args2 ?: ''
+    def args3 = task.ext.args3 ?: ''
+    def args4 = task.ext.args4 ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = "1.15" // Staden_io versions break the pipeline
+    """
+    cram_filter -n ${from}-${to} ${cramfile} - | \\
+        samtools fastq ${args1} - |  \\
+        minimap2 -t${task.cpus} -R '${rglines}' ${args2} ${ref} - |  \\
+        ${projectDir}/bin/grep_pg.sh |  \\
+        perl ${projectDir}/bin/filter_five_end.pl |  \\
```

Comment:
Is there a reason why 5' filtering is not happening when [mapping is done with bwa-mem2](https://github.com/sanger-tol/genomeassembly/blob/hic_minimap2/modules/local/cram_filter_align_bwamem2_fixmate_sort.nf
) and is happening here for minimap2?

Reply:
5' removal happens within bwamen2 as I remember. Not sure need to look up.